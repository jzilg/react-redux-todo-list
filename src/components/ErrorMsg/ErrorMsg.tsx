import React, { ReactElement } from 'react'
import style from './error-msg.scss'

interface ErrorProps {
    message: string
}

const ErrorMsg = ({ message }: ErrorProps): ReactElement<{}> => (
    <div className="container">
        <div className="error">
            <h2 className={style.title}>
                {message}
            </h2>
        </div>
    </div>
)

export default ErrorMsg
