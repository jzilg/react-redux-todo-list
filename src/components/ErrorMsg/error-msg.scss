@import "../../style/color";

.title {
    color: $red;
}
