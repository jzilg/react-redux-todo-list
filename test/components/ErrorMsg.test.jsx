import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { shallowToJson } from 'enzyme-to-json'
import expect from 'expect'
import ErrorMsg from '../../src/components/ErrorMsg'

Enzyme.configure({ adapter: new Adapter() })

describe('ErrorMsg', () => {
    it('should render correctly', () => {
        const message = 'An ErrorMsg occurred'
        const component = shallow(<ErrorMsg message={message} />)
        const tree = shallowToJson(component)
        expect(tree).toMatchSnapshot()
    })
})
