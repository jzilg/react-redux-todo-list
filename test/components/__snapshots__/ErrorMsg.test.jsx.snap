// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMsg should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="error"
  >
    <h2>
      An ErrorMsg occurred
    </h2>
  </div>
</div>
`;
